### PR TITLE
Add glossary entries for keyword and meta-schema.

### PR DESCRIPTION
**Pull request by Julian**
Also loosens the language around schema<->instance interaction from https://github.com/json-schema-org/json-schema-org.github.io/pull/462#issuecomment-1228956451.

### CODE REVIEW

**Review by handrews — APPROVED** [review]
Nice, thanks!

File: learn/glossary.md [hunk 1]
```diff
@@ -24,15 +24,30 @@ While future drafts may introduce new behavior or changes to existing behavior,
...
 
 The JSON Schema specification makes no broad assumptions about the structure of instances themselves beyond those of the JSON specification itself.
 In particular it does not reserve any properties within a JSON object for its own use, or require parsers of JSON to support features beyond those already mandated of JSON implementations.
 
+### meta-schema
+
+A [schema](#schema) which is itself intended to describe other *schemas*.
+
+JSON Schema defines a language for describing any [instance](#instance) using a schema written in JSON.
+Since schemas are themselves JSON values, they may be opaquely treated as *instances*, and therefore described by other schemas.
```
Comment (handrews) [inline learn/glossary.md:47]: Actually it's not entirely opaque, §9.3.3 notes that in some circumstances an alternate way of handling schema-instances SHOULD be implemented.  I'm not quite sure how that should be dealt with as we don't want a huge tangent on an obscure case, but it should be acknowledged that meta-schema validation can be more complex.

**Review by handrews — CHANGES_REQUESTED** [review]

File: learn/glossary.md (hunk 1 above)
Comment (Julian) [reply learn/glossary.md:47]: Are you happy with deleting the word `opaquely` and just saying `may also be treated as instances`? That matches the language from [§4.3](https://json-schema.org/draft/2020-12/json-schema-core.html#name-json-schema-documents).

Comment (handrews) [reply learn/glossary.md:47]: @Julian yeah that would work, thanks!  Better than too much detail on the basic concept.  I'll bonk the approve button again when I see that change, but feel free to consider it pre-approved - everything else is great.

Comment (Julian) [reply learn/glossary.md]: Great, done. Thanks.

**Review by handrews — APPROVED** [review]

**Pull request merged**